src/utils/domains.ts

```diff
@@ -1,8 +1,17 @@
 import { Id } from '@colony/colony-js';
 
 import { COLONY_TOTAL_BALANCE_DOMAIN_ID } from '~constants';
+import { Domain } from '~gql';
+import { Colony } from '~types';
+
+import { notNull } from './arrays';
 
 export const getDomainId = (ethDomainId?: number) =>
   ethDomainId === COLONY_TOTAL_BALANCE_DOMAIN_ID || ethDomainId === undefined
     ? Id.RootDomain
     : ethDomainId;
+
+export const getDomain = (colony: Colony, domainId: number) =>
+  colony?.domains?.items
+    .filter(notNull)
+    .find(({ nativeId }) => nativeId === domainId) as Domain;
```

Comment:
I think this casting is unnecessary and actually may lead to some issues:
- `find` isn't guaranteed to return an item, so it might return `undefined`. The casting changes the return type of this function to `Domain`.
- `Domain`, as exported from `~gql`, describes the GraphQL `Domain` model defined in the schema. It is not the same as the subset of fields we select from `colony.domains.items`, in fact we're losing the renamed field as a result of that:
`Domain` type shape:
```
{
  __typename?: 'Domain';
  colonyDomainsId?: Maybe<Scalars['ID']>;
  color?: Maybe<DomainColor>;
  createdAt: Scalars['AWSDateTime'];
  description?: Maybe<Scalars['String']>;
  domainParentId?: Maybe<Scalars['ID']>;
  id: Scalars['ID'];
  name?: Maybe<Scalars['String']>;
  nativeId: Scalars['Int'];
  parent?: Maybe<Domain>;
  updatedAt: Scalars['AWSDateTime'];
};
```

The subset of fields we're querying for a colony:
```
 {
      color
      description
      id
      name
      nativeId
      parentId: domainParentId
    }
```

I suggest that you create a dedicated fragment on `Domain`, which you can then export from `types/graphql` similarly to other fragments

Reply:
Nice suggestion